### PR TITLE
Unit test

### PR DESCRIPTION
**Pull request by frozen599**

### CODE REVIEW

File: api/cmd/friends/main.go [hunk 1]
```diff
@@ -10,12 +11,15 @@ import (
...
-	cfg := config.NewConfig()
+	cfg := config.NewConfig("./../../..")
+	log.Println(cfg)
 	db := config.InitDB(cfg)
+	log.Println(db == nil)
 	if db == nil {
 		log.Fatal("cannot establish connection to db")
 	}
 	defer db.Close()
+	fmt.Println(cfg)
```
Comment (nhan1603) [inline api/cmd/friends/main.go]: remove this  print command as the configuration will be printed on the log

File: api/internal/controller/subscriber.go [hunk 2]
```diff
@@ -34,57 +34,88 @@ func (sc subscriberController) CreateSubScription(requestor, target string) erro
...
+
+	blockingRelationShip := models.Relationship{
+		UserID1:          requestorUser.ID,
+		UserID2:          targetUser.ID,
+		RelationshipType: models.RelationshipTypeSubscriber,
+		CreatedAt:        time.Now(),
+		UpdatedAt:        time.Now(),
+	}
+
+	err = sc.relaRepo.CreateRelationship(blockingRelationShip)
```
Comment (nhan1603) [inline api/internal/controller/subscriber.go:65]: can we return the result from repository immediately ?

File: api/internal/handler/subscriber_handler_test.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	cfg := config.NewConfig("./../../..")
+	dbInstance := config.InitDB(cfg)
+	defer dbInstance.Close()
+	//initData, err := ioutil.ReadFile("./test_data/init_data.sql")
+	//require.NoError(t, err)
+	//_, err = dbInstance.Exec(string(initData))
+	//require.NoError(t, err)
+	//deleteData, err := ioutil.ReadFile("./test_data/delete_data.sql")
+	//require.NoError(t, err)
+	//defer dbInstance.Exec(string(deleteData))
```
Comment (nhan1603) [inline api/internal/handler/subscriber_handler_test.go:117]: Commented LOCs can be removed.

File: api/internal/controller/test_data/delete_data.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+DELETE FROM relationships;
+
```
Comment (nhan1603) [inline api/internal/controller/test_data/delete_data.sql:2]: Remove empty line

File: api/internal/pkg/response.go [hunk 5]
```diff
@@ -30,13 +30,11 @@ func ResponseError(w http.ResponseWriter, code int, err error) {
 	resp := make(map[string]interface{})
 	resp["success"] = false
 	resp["message"] = err.Error()
-	if code == http.StatusInternalServerError {
-		resp["message"] = "INTERNAL SERVER ERROR"
-	}
 	respData, err := json.Marshal(resp)
 	if err != nil {
 		log.Fatal(err)
```
Comment (nhan1603) [inline api/internal/pkg/response.go]: Why should we stop the program when the writer cannot serialize the response ?

**Review by nhan1603 — CHANGES_REQUESTED** [review]
Please perform similar checks for all the commented issues.

File: api/internal/pkg/response.go [hunk 6]
```diff
@@ -16,7 +16,7 @@ func ResponseOk(w http.ResponseWriter) {
 	}
 	respData, err := json.Marshal(&successResp)
 	if err != nil {
-		log.Fatal(err)
+		fmt.Println(err)
```
Comment (nhan1603) [inline api/internal/pkg/response.go:19]: please return here and process the response writer to write an error message for user to understand

**Review by nhan1603 — CHANGES_REQUESTED** [review]

**Review by nhan1603 — APPROVED** [review]
LGTM

File: api/cmd/friends/main.go (hunk 1 above)
Comment (frozen599) [reply api/cmd/friends/main.go]: resolved in 4159eeb1fb0894dc0652209c0914bc5d301eb413

**Pull request merged**